packages/core/src/__integrationtests__/Deposit.spec.ts

```diff
@@ -0,0 +1,489 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @group integration/deposit
+ */
+
+import {
+  createLightDidFromSeed,
+  createOnChainDidFromSeed,
+  DemoKeystore,
+  DidChain,
+  FullDidDetails,
+  SigningAlgorithms,
+} from '@kiltprotocol/did'
+import {
+  IRequestForAttestation,
+  KeyRelationship,
+  KeystoreSigner,
+  SubmittableExtrinsic,
+} from '@kiltprotocol/types'
+import { Keyring } from '@kiltprotocol/utils'
+import { KeyringPair } from '@polkadot/keyring/types'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+import { mnemonicGenerate, randomAsHex } from '@polkadot/util-crypto'
+import {
+  getDidDeposit,
+  createAttestation,
+  getAttestationDeposit,
+  createMinimalFullDidFromLightDid,
+  WS_ADDRESS,
+  devFaucet,
+  DriversLicense,
+  initialTransfer,
+  CtypeOnChain,
+} from './utils'
+import { Balance } from '../balance'
+import Attestation from '../attestation/Attestation'
+import Claim from '../claim/Claim'
+import RequestForAttestation from '../requestforattestation/RequestForAttestation'
+import { disconnect, init } from '../kilt'
+
+let tx: SubmittableExtrinsic
+let authorizedTx: SubmittableExtrinsic
+let storedEndpointsCount: number
+
+async function checkDeleteFullDid(
+  identity: KeyringPair,
+  fullDid: FullDidDetails,
+  keystore: DemoKeystore
+): Promise<boolean> {
+  storedEndpointsCount = await DidChain.queryEndpointsCounts(fullDid.did)
+  const deleteDid = await DidChain.getDeleteDidExtrinsic(storedEndpointsCount)
+
+  const refreshedTxIndex = await fullDid.refreshTxIndex()
+
+  tx = await DidChain.generateDidAuthenticatedTx({
+    didIdentifier: identity.address,
+    txCounter: refreshedTxIndex.getNextTxIndex(),
+    call: deleteDid,
+    signer: keystore as KeystoreSigner<string>,
+    signingPublicKey: fullDid.getKeys(KeyRelationship.authentication)[0]
+      .publicKeyHex,
+    alg: fullDid.getKeys(KeyRelationship.authentication)[0].type,
+    submitter: identity.address,
+  })
+
+  const balanceBeforeDeleting = await Balance.getBalances(
+    identity.address
+  ).then((balance) => balance)
+
+  const didDeposit = await getDidDeposit(identity.address)
+
+  await BlockchainUtils.signAndSubmitTx(tx, identity, {
+    resolveOn: BlockchainUtils.IS_FINALIZED,
+  })
+
+  const balanceAfterDeleting = await Balance.getBalances(identity.address).then(
+    (balance) => balance
+  )
+
+  if (
+    balanceBeforeDeleting.reserved.toNumber() - didDeposit.toNumber() ===
```

Comment:
`.toNumber()` is discouraged to be used with `BN` for potential overflows. I would simply subtract two `BN`s together, and then simply compare for equality with `balanceAfterDeleting.reserved`. Same in the other places.